hardhat.config.js

```diff
@@ -47,17 +50,31 @@ module.exports = {
     },
     hardhat:{
       allowUnlimitedContractSize:false,
-      accounts:walletUtils.localWallet("1000000000000000000000",num=20),
+      gas: 6000000,
+      accounts:walletUtils.localWallet("1000000000000000000000000",num=20),
       forking : {
-        //url:`https://mainnet.infura.io/v3/${infuraKey}`
-        url:`https://eth-mainnet.alchemyapi.io/v2/${alchemyKey}`
+      //   url:`https://mainnet.infura.io/v3/${infuraKey}`
+      url:`https://eth-mainnet.alchemyapi.io/v2/${alchemyKey}`
       }
     },
     kovan:{
-      url:`https://kovan.infura.io/v3/${infuraKey}`,
+      //url:`https://kovan.infura.io/v3/${infuraKey}`,
+      url:"https://eth-kovan.alchemyapi.io/v2/DvW1I4OgMAVXJIw3zzfWHnQz1Lpeki9I",
       accounts:walletUtils.makeKeyList(),
       chainId:42,
-      gas: 12500000,
+      gas: 1250000,
+      gasMultiplier:2
+    },
+    maticMumbai: {
+      url:"https://rpc-mumbai.matic.today",
+      accounts:walletUtils.makeKeyList(),
+      chainId: 80001
+    },
+    rinkeby: {
+      url: `https://rinkeby.infura.io/v3/d126f392798444609246423b06116c77`,
```

Comment:
remove infura API

Reply:
It accepts both keys so I have just removed hardcoded stuff